torch/distributed/nn/api/remote_module.py

```diff
@@ -422,10 +422,10 @@ def named_modules(
         _raise_not_supported(self.named_modules.__name__)
 
     def train(self: T, mode: bool = True) -> T:  # type: ignore[return]
-        _raise_not_supported(self.train.__name__)
+        return self.module_rref.rpc_sync().train()
```

Comment:
This will make `train()` a blocking op. Not sure if we need to have a separate async version.

@pbelevich Do you think we need an async version? I am not sure if a local host always wants to wait for the training on a remote worker.